Samples/Notifications/Push/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,160 @@
+using namespace winrt::Windows::Foundation;
+
+// "To obtain your Azure AppId,
+// follow "Configure your app's identity in Azure Active Directory" at https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/notifications/push/push-quickstart
+winrt::guid remoteId{ "00000000-0000-0000-0000-000000000000"}; // Replace this with your own Azure AppId
+
+winrt::Windows::Foundation::IAsyncOperation<PushNotificationChannel> RequestChannelAsync()
+{
+    auto channelOperation{ PushNotificationManager::Default().CreateChannelAsync(remoteId) };
+
+    // Setup the in-progress event handler
+    channelOperation.Progress(
+        [](auto&& sender, auto&& args)
+        {
+            if (args.status == PushNotificationChannelStatus::InProgress)
+            {
+                // This is basically a noop since it isn't really an error state
+                std::cout << "\nWNS Channel URI request is in progress." << std::endl;
+            }
+            else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+            {
+                LOG_HR_MSG(
+                    args.extendedError,
+                    "The WNS Channel URI request is in back-off retry mode because of a retryable error! Expect delays in acquiring it. RetryCount = %d",
+                    args.retryCount);
+            }
+        });
+
+    auto result{ co_await channelOperation };
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        auto channel{ result.Channel() };
+
+        std::cout << "\nWNS Channel URI: " << winrt::to_string(channel.Uri().ToString()) << std::endl;
+
+        // It's the caller's responsibility to keep the channel alive
+        co_return channel;
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with the WNS Channel URI request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task{ RequestChannelAsync() };
+    if (task.wait_for(std::chrono::minutes(5)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result{ task.GetResults() };
+    return result;
+}
+
+// Subscribe to an event which will get signaled whenever a foreground notification arrives.
+void SubscribeForegroundEventHandler()
+{
+    winrt::event_token token{ PushNotificationManager::Default().PushReceived([](auto const&, PushNotificationReceivedEventArgs const& args)
+    {
+        auto payload{ args.Payload() };
+
+        std::string payloadString(payload.begin(), payload.end());
+        std::cout << "\nPush notification content received in the FOREGROUND: " << payloadString << std::endl;
```

Comment:
Probably want to get rid of the \n if the prior console input used std::endl.

Reply:
Recommend: stick to endl or \n. Not both.

Reply:
\n and endl perform different actions. \n simple creates a new line while endl flushes all buffers (which also generates a newline)

the \n is intentional. It creates some breathing room to make the text easier to read. The endl ensure the buffers get flushed and all text is visible on screen.

The first version of the sample was using two endl between some outputs to separate things and make the output easier to read. I received feedback that flushing the buffer twice wasn't a good idea. I also want to make sure the buffers get flushed once in a while as it helps ensure all output is visible in case of a bug / crash.

Reply:
config/gni/devtools_grd_files.gni
front_end/models/bindings/BUILD.gn
front_end/models/formatter/BUILD.gn
front_end/models/source_map_scopes/BUILD.gn
front_end/models/source_map_scopes/NamesResolver.ts
front_end/models/source_map_scopes/source_map_scopes.ts
front_end/models/workspace/BUILD.gn
front_end/panels/sources/BUILD.gn
front_end/panels/sources/DebuggerPlugin.ts
front_end/panels/sources/ScopeChainSidebarPane.ts
front_end/panels/sources/sources-legacy.ts
front_end/panels/sources/sources.ts

Reply:
config/gni/devtools_grd_files.gni
front_end/core/host/UserMetrics.ts
front_end/core/i18n/locales/en-US.json
front_end/core/i18n/locales/en-XL.json
front_end/entrypoints/lighthouse_worker/LighthouseService.ts
front_end/entrypoints/main/MainImpl.ts
front_end/panels/lighthouse/BUILD.gn
front_end/panels/lighthouse/LighthouseController.ts
front_end/panels/lighthouse/LighthousePanel.ts
front_end/panels/lighthouse/LighthouseProtocolService.ts
front_end/panels/lighthouse/LighthouseStartView.ts
front_end/panels/lighthouse/LighthouseStartViewFR.ts